ddtrace/profiling/serverless_scheduler.py

```diff
@@ -0,0 +1,31 @@
+# -*- encoding: utf-8 -*-
+import logging
+
+import attr
+
+from ddtrace.internal import compat
+from ddtrace.internal.utils import attr as attr_utils
+from ddtrace.profiling import scheduler
+
+
+LOG = logging.getLogger(__name__)
+
+
+@attr.s
+class ServerlessScheduler(scheduler.Scheduler):
+    _interval = attr.ib(factory=attr_utils.from_env("DD_SERVERLESS_PROFILING_UPLOAD_INTERVAL", 1, int))
+    _total_profiled_seconds = attr.ib(default=0)
+
+    def periodic(self):
+        now = compat.time_ns()
+        # Guard against _last_export not being set
+        last_export = self._last_export or compat.time_ns()
+        if now - last_export >= 60 * 1e9 and self._total_profiled_seconds >= 60:
+            self._total_profiled_seconds = 0
+            start_time = compat.monotonic()
+            try:
+                self.flush()
+            finally:
+                self.interval = max(0, self._configured_interval - (compat.monotonic() - start_time))
+        else:
+            self._total_profiled_seconds += 1
```

Comment:
```suggestion
            self._total_profiled_seconds += self.interval
```
but you don't need this, you can just compute `time.now() - self._last_export` and see if it's `>= 60`)

Reply:
I had tried that initially, but it won't work because Lambda functions are frozen in between invocations. So `_last_export` could be 20 minutes ago, even though we've only profiled 1 or two invocations totaling a few milliseconds.

Reply:
Resolved by just checking last export in combination with a wakeup of once per 60s